lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1868,8 +1868,11 @@ static LogicalResult foldHiddenReset(RegOp reg, PatternRewriter &rewriter) {
   }
   auto pt = rewriter.saveInsertionPoint();
   rewriter.setInsertionPoint(con);
-  replaceOpWithNewOpAndCopyName<ConnectOp>(rewriter, con, con.getDest(),
-                                           constReg ? constOp : mux.getLow());
+  replaceOpWithNewOpAndCopyName<ConnectOp>(
+      rewriter, con, con.getDest(),
+      constReg
+          ? static_cast<mlir::TypedValue<FIRRTLBaseType>>(constOp.getResult())
+          : mux.getLow());
```

Comment:
If anyone has any suggestions for how to do this more cleanly, please let me know. My C++ fu with regards to inferred type conversions isn't good enough to come up with something better. Without my changes here, I get the error:

```
error: conditional expression is ambiguous; 'TypedValue<::circt::firrtl::IntType>' can be converted to 'TypedValue<::circt::firrtl::FIRRTLBaseType>' and vice versa
```

I tried breaking out the ternary expression into its own variable with an explicit type declared, but that apparently wasn't good enough for it to infer the desired conversion. I had something like:

```c++
  mlir::TypedValue<FIRRTLBaseType>> newOp = constReg
          ? constOp.getResult()
          : mux.getLow();
  replaceOpWithNewOpAndCopyName<ConnectOp>(
      rewriter, con, con.getDest(),
      newOp);
```

but it still has the same ambiguous conversion issue for some reason.

Reply:
Can you not just `(Value)` cast from `constOp` and/or `mux.getLow()`?

Reply:
Yeah, that looks like it works, thanks. I'm still not experienced enough with C++ to understand when to use different kinds of casts, but is this a situation where `static_cast<Value>()` would be preferred over the C-style cast, or is the extra verbosity not worth it here?

Reply:
Since in this case you're merely clarifying to the compiler which type the result should be, it's not worth the additional verbosity. Ideally the compiler would be smart enough to magically figure this out, but there's likely some sort of ambiguity introduced by the `TypedValue` subclass.

Reply:
Gotcha, sounds good. I amended my commit to use the simple cast you suggested.